tools/oeutil/host/generate_evidence.cpp

```diff
@@ -445,56 +448,117 @@ static const oe_claim_t* find_claim(
-            printf("%02x", claims[i].value[j]);
-        printf("\n\n");
-    }
-    // tcb status
-    printf("claims[%zu]: %s\n%u\n\n", i, claims[i].name, *claims[i].value);
-    i++;
-    // tcb date, validity
-    for (; i < 21; i++)
-    {
-        printf("claims[%zu]: %s\n", i, claims[i].name);
-        uint32_t* date = (uint32_t*)claims[i].value;
-        for (size_t j = 0; j < 6; j++)
-            printf("%d ", date[j]);
-        printf("\n\n");
+        case 'u':
+        {
+            printf(
+                "claims[%zu]: %s\n%u\n\n",
+                claim_number,
+                claim_name,
+                *claim_value);
+            break;
+        }
+        case 'x':
+        {
+            printf(
+                "claims[%zu]: %s (%zu)\n0x",
+                claim_number,
+                claim_name,
+                claim_value_size);
+            for (size_t j = 0; j < claim_value_size; j++)
+                printf("%02x", claim_value[j]);
+            printf("\n\n");
+            break;
+        }
+        case 's':
+        {
+            printf(
+                "claims[%zu]: %s\n%s\n\n",
+                claim_number,
+                claim_name,
+                claim_value);
+            break;
+        }
+        case 'D':
+        {
+            printf("claims[%zu]: %s\n", claim_number, claim_name);
+            uint32_t* date = (uint32_t*)claim_value;
+            for (size_t j = 0; j < 6; j++)
+                printf("%d ", date[j]);
+            printf("\n\n");
+            break;
+        }
+        default:
+        {
+            printf("The claim format is not recognised.\n");
+        }
     }
-    // sgx pcesvn, qeid, and fmspc
-    for (; i < 24; i++)
-    {
-        printf(
-            "claims[%zu]: %s (%zu)\n0x",
+}
+
+void dump_claims(const oe_claim_t* claims, size_t claims_length)
+{
+    printf("\n%zu OE claims retrieved:\n\n", claims_length);
+
+    /*
+     * This map holds a mapping to claims and their expected
+     * print format strings and needs to be updated when a
+     * new claim is added. The convention followed is:
+     * %u   : 'u'
+     * %02x : 'x'
+     * %s   : 's'
+     * date : 'D'
+     */
+    std::map<std::string, char> claims_format{
+        {"id_version", 'u'},
+        {"security_version", 'u'},
+        {"attributes", 'u'},
+        {"unique_id", 'x'},
+        {"signer_id", 'x'},
+        {"product_id", 'x'},
+        {"format_uuid", 'x'},
+        {"sgx_pf_gp_exit_info_enabled", 'x'},
+        {"sgx_isv_extended_product_id", 'x'},
+        {"sgx_is_mode64bit", 'x'},
+        {"sgx_has_provision_key", 'x'},
+        {"sgx_has_einittoken_key", 'x'},
+        {"sgx_uses_kss", 'x'},
+        {"sgx_config_id", 'x'},
+        {"sgx_config_svn", 'x'},
+        {"sgx_isv_family_id", 'x'},
+        {"sgx_cpu_svn", 'x'},
+        {"tcb_status", 'u'},
+        {"tcb_date", 'D'},
+        {"validity_from", 'D'},
+        {"validity_until", 'D'},
+        {"ue_id", 'x'},
```

Comment:
is there any order to this list of entries?  Seems like alphabetical by name would make more sense.

Reply:
I have listed them in the order they appear on the claims buffer and in the Design Doc - https://github.com/rrathna/openenclave/blob/update_claim_tags/docs/DesignDocs/RemoteAttestationCollaterals.md

Reply:
What happens if the same map key appears twice?  Unless it generates a build error, then this order is bug prone.

Reply:
Re-ordered to be alphabetical